### ISSUE TITLE
setting json to a firestore document ends up with seg fault when it is built with a proto dependency

### ISSUE THREAD

**Issue by cdg720** [googleapis/python-firestore#19]
#### Environment details

  - OS type and version: `Linux version 4.19.87+ (chrome-bot@chromeos-ci-legacy-us-east1-d-x32-66-58ml) (Chromium OS 10.0_pre370808_p20191029-r5 clang version 10.0.0` (cloudshell)
  - Python version: `Python 3.7.3`
  - pip version:  `pip 19.0.3 from /usr/local/lib/python3.7/site-packages/pip (python 3.7)`
  - `google-cloud-firestore` version: `Version: 1.6.2`

#### Steps to reproduce
```bash
git clone https://github.com/cdg720/bazel_examples.git
cd firestore
bazel build -c opt foo:bar
# This ends up with segmentation fault.
./bazel-bin/foo/bar \
  --key credential_key \
  --project project_id \
  --test=test
```

```bash
# The following runs fine.
python3 foo/bar.py \
  --key credential_key \
  --project project_id 
  --test=test
```

#### Code example

```python
# foo/bar.py
import os
from google.cloud import firestore

from absl import app
from absl import flags

FLAGS = flags.FLAGS

flags.DEFINE_string('key', None, 'key')
flags.DEFINE_string('project', None, 'project')
flags.DEFINE_string('test', None, 'test')


def main(argv):
  os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = FLAGS.key

  db = firestore.Client(project=FLAGS.project)
  doc_ref = db.collection('debug').document(FLAGS.test)
  data = {'id': doc_ref.id, 'name': doc_ref.id}

  doc_ref.set(data)


if __name__ == "__main__":
  app.run(main)
```

```BUILD
# foo/BUILD
py_binary(
    name = "bar",
    srcs = ["bar.py"],
    deps = [
        "//protos:empty_py",
    ],
)
```

```BUILD
# protos/BUILD
package(default_visibility = [
    "//foo:__pkg__",
])

load("@build_stack_rules_proto//python:python_proto_library.bzl", "python_proto_library")
load("@rules_proto//proto:defs.bzl", "proto_library")

# empty.proto is an empty file.
proto_library(
    name = "empty_proto",
    srcs = ["empty.proto"],
)

python_proto_library(
    name = "empty_py",
    deps = [ ":empty_proto" ],
)
```

```WORKSPACE
# WORKSPACE
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

http_archive(
    name = "build_stack_rules_proto",
    urls = ["https://github.com/stackb/rules_proto/archive/b93b544f851fdcd3fc5c3d47aee3b7ca158a8841.tar.gz"],
    sha256 = "c62f0b442e82a6152fcd5b1c0b7c4028233a9e314078952b6b04253421d56d61",
    strip_prefix = "rules_proto-b93b544f851fdcd3fc5c3d47aee3b7ca158a8841",
)

load("@build_stack_rules_proto//python:deps.bzl", "python_proto_library")

python_proto_library()

load("@io_bazel_rules_python//python:pip.bzl", "pip_import", "pip_repositories")

pip_repositories()

pip_import(
    name = "protobuf_py_deps",
    requirements = "@build_stack_rules_proto//python/requirements:protobuf.txt",
)

load("@protobuf_py_deps//:requirements.bzl", protobuf_pip_install = "pip_install")

protobuf_pip_install()
```

#### Stack trace
```
Fatal Python error: Segmentation fault
Thread 0x00007f0012ffd700 (most recent call first):
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 78 in _worker
  File "/usr/local/lib/python3.7/threading.py", line 865 in run
  File "/usr/local/lib/python3.7/threading.py", line 917 in _bootstrap_inner
  File "/usr/local/lib/python3.7/threading.py", line 885 in _bootstrap
Thread 0x00007f00210fd500 (most recent call first):
  File "/usr/local/lib/python3.7/site-packages/grpc/_channel.py", line 813 in _blocking
  File "/usr/local/lib/python3.7/site-packages/grpc/_channel.py", line 825 in __call__
  File "/usr/local/lib/python3.7/site-packages/google/api_core/grpc_helpers.py", line 57 in error_remapped_callable
  File "/usr/local/lib/python3.7/site-packages/google/api_core/timeout.py", line 214 in func_with_timeout
  File "/usr/local/lib/python3.7/site-packages/google/api_core/retry.py", line 184 in retry_target
  File "/usr/local/lib/python3.7/site-packages/google/api_core/retry.py", line 286 in retry_wrapped_func
  File "/usr/local/lib/python3.7/site-packages/google/api_core/gapic_v1/method.py", line 143 in __call__
  File "/usr/local/lib/python3.7/site-packages/google/cloud/firestore_v1/gapic/firestore_client.py", line 1033 in commit
  File "/usr/local/lib/python3.7/site-packages/google/cloud/firestore_v1/batch.py", line 147 in commit
  File "/usr/local/lib/python3.7/site-packages/google/cloud/firestore_v1/document.py", line 234 in set
  File "/home/admin_/bazel_examples/firestore/./bazel-bin/foo/bar.runfiles/__main__/foo/bar.py", line 21 in main
  File "/usr/local/lib/python3.7/site-packages/absl/app.py", line 250 in _run_main
  File "/usr/local/lib/python3.7/site-packages/absl/app.py", line 299 in run
  File "/home/admin_/bazel_examples/firestore/./bazel-bin/foo/bar.runfiles/__main__/foo/bar.py", line 25 in <module>
```

I am not sure if this is a google-cloud-firestore problem or a bazel problem or a build_stack_rules_proto problem.

It only fails on linux machines and runs fine on macs.

Thanks,
DK

**Comment by crwilcox** [comment]
@cdg720 we are nearing a release of a new version of the library (v2.0.0). This uses an overhauled proto layer, so it is quite possible this is no longer going to repro. If you have a chance to try that out do let us know. It is currently available as 2.0.0-dev2, but I expect in the next couple of days to GA 2.0.0.

**Comment by tseaver** [comment]
Closing on the presumption that the 2.0.0 release fixed the issue.  Please follow-up if you can still reproduce.
